File: code-examples/php/indexes/searchIndexes.php
<?php

require 'vendor/autoload.php';

// Search Index operations

// Replace the following with your MongoDB deployment's connection string
$uri = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";

$client = new MongoDB\Client($uri);
$database = $client->selectDatabase("<databaseName>");
$collection = $database->selectCollection("<collectionName>");

// start createSearchIndex example
// Create a search index
$index1 = [
    'name' => 'search1',
    'definition' => [
        'mappings' => [
            'dynamic' => true,
        ],
    ],
];

$collection->createIndex($index1);
// end createSearchIndex example

// start listSearchIndexes example
// List search indexes
$indexes = $collection->listIndexes();
echo "Existing search indexes:\n";
foreach ($indexes as $idx) {
    var_dump($idx);
}
// end listSearchIndexes example

// start updateSearchIndex example
// Update a search index
$index2 = [
    'mappings' => [
        'dynamic' => true,
        'fields' => [
            'description' => [
                'type' => 'string'
            ],
        ],
    ]
];

$collection->dropIndex('search1');
$collection->createIndex($index2, ['name' => 'search1']);
// end updateSearchIndex example

// start dropSearchIndex example
// Dropping (deleting) a search index
$collection->dropIndex('search1');
// end dropSearchIndex example

File: code-examples/scala/indexes/searchIndexes.scala
import org.mongodb.scala._
import org.mongodb.scala.bson.BsonDocument
import scala.concurrent.{Await, Future}
import scala.concurrent.duration._
import scala.util.{Failure, Success}

// Search Index operations 
object SearchIndexManager {

  val mongoClient: MongoClient = MongoClient("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority")

  val database: MongoDatabase = mongoClient.getDatabase("<databaseName>")
  val collection: MongoCollection[Document] = database.getCollection("<collectionName>")

  // Create a search index
  val index1: BsonDocument = BsonDocument("name" -> "search1",
    "definition" -> BsonDocument("mappings" -> BsonDocument("dynamic" -> true)))
  createSearchIndex(index1)

  // List search indexes
  listSearchIndexes()

  // Update a search index
  val index2: BsonDocument = BsonDocument("mappings" -> BsonDocument("dynamic" -> true, "fields" ->
    BsonDocument("description" -> BsonDocument("type" -> "string"))))
  updateSearchIndex("search1", index2)

  // Dropping (deleting) a search index
  dropSearchIndex("search1")

  mongoClient.close()

  def createSearchIndex(index: BsonDocument): Unit = {
    collection.createIndex(index)
  }

  def listSearchIndexes(): Unit = {
    val listSearchIndexes: Future[Seq[String]] = collection.listIndexes().map( doc => doc.toJson()).toFuture()
    listSearchIndexes onComplete {
      case Success(searchIndexes) => println("Existing search indexes:\n" + searchIndexes.mkString("\n"))
      case Failure(e) => println("Failed with: " + e)
    }
  }

  def updateSearchIndex(name: String, index: BsonDocument): Unit = {
    collection.dropIndex(name)
    collection.createIndex(index)
  }

  def dropSearchIndex(name: String): Unit = {
    collection.dropIndex(name)
  }
}

// Run the SearchIndexManager object
SearchIndexManager
